projects/map-ol/src/lib/map-ol.service.ts

```diff
@@ -1444,13 +1463,21 @@ export class MapOlService {
     let popupHtml = '';
     if (args.popupFn) {
       popupHtml = args.popupFn(popupObj);
-    }
-    else if (html && (!popupObj || Object.keys(popupObj).length === 0)) {
+    } else if (html && (!popupObj || Object.keys(popupObj).length === 0)) {
       popupHtml = html;
     } else {
       popupHtml = this.createPopupHtml(popupObj);
     }
     content.innerHTML = popupHtml;
+    if (args.dynamicPopup) {
+      // To prevent memory leak:
+      // if this very popup already has been created (for example `popup_move_ID`),
+      // then destroy it before creating a new one.
+      const id = overlay.getId().toString();
+      this.destroyDynamicPopupComponent(id);
+      // Only now create a new one.
+      this.createDynamicPopupComponent(id, content, args);
```

Comment:
Here we could argue about passing the `popupObj` too and use it to get the properties.

- An enhancement would be, that it then would work with `popup['filterkeys']` and `popup['properties']` from the [type popup](https://github.com/dlr-eoc/ukis-frontend-libraries/blob/v7.2.0/projects/services-layers/src/lib/types/Layers.ts#L31)
- A disadvantage is that you dont get all the properties from `IPopupArgs` so you could not use the layer or feature from it in the popup if you really need this. Maybe we could use here a extra type e.g.
```ts
IDynamicPopupArgs:{
  layer: IPopupArgs['layer'];
  feature?: IPopupArgs['feature'];
  properties: popupObj
  dynamicPopup?: popup['dynamicPopup']; // or  IPopupArgs['dynamicPopup'];
}

createDynamicPopupComponent(id: string, anchorElement: HTMLElement, args: IDynamicPopupArgs);
```


```ts
...
this.destroyDynamicPopupComponent(id);
// Only now create a new one.
const dargs: IDynamicPopupArgs = {
    layer: args.layer
    feature?: args.feature || null
    properties: popupObj
    dynamicPopup: args.dynamicPopup
}
this.createDynamicPopupComponent(id, content, dargs);
```

Reply:
Thanks a lot!
Just to clarify: 
In `properties: popupObj`, did you mean that the field `properties` should be of the type `@dlr-eoc/services-layers.popup`, or of the same type as the variable `popupObj` that is passed to `addPopup`?

I'm assuming it's the former - because the latter can also be [just a string](https://github.com/dlr-eoc/ukis-frontend-libraries/blob/e35dd0354551f2ae9dbfd3f7b69782c2768a98a2/projects/map-ol/src/lib/map-ol.service.ts#L1346) - correct? (I think that this variable `popupObj` has been used for a few different things over time...)